scapy/sendrecv.py

```diff
@@ -616,19 +616,20 @@ def _parse_tcpreplay_result(stdout_b, stderr_b, argv):
 def _interface_selection(iface,  # type: Optional[_GlobInterfaceType]
                          packet  # type: _PacketIterable
                          ):
-    # type: (...) -> _GlobInterfaceType
+    # type: (...) -> Tuple[NetworkInterface, bool]
     """
     Select the network interface according to the layer 3 destination
     """
-
+    _iff, src, _ = next(packet.__iter__()).route()
+    ipv6 = bool(src and ":" in src)
```

Comment:
I will advise to use `socket.inet_pton(socket.AF_INET6, src)` to make the test more robust.